contrib/babelfishpg_unit/test_ssl_write.c

```diff
@@ -0,0 +1,108 @@
+#include "babelfishpg_unit.h"
+#include "../babelfishpg_tds/src/include/tds_secure.h"
+
+#ifdef USE_SSL
+static char *prelogin; 
+
+static ssize_t 
+mock_socket_write(Port *port, const void *ptr, size_t len)
+{
+
+    /*
+     *  Mock function of tds_secure_raw_write() present in tdssecure.c file
+     */
+
+    const unsigned char *ptr_u8 = (const unsigned char *) ptr;
+    signed char *prelogin_ptr = (signed char *) prelogin;
+    int i;
+    for (i = 0; i < len; i++) 
+        sscanf((const char *) &ptr_u8[i * 2], "%2hhx", (signed char *) &prelogin_ptr[i]);
+    return len;
+}
+
+
+TestResult*
+test_ssl_handshakeWrite(void)
+{
+
+    /*
+     *  We will generate a message and pass it to the handshake write function
+     *  We will be comparing the number of bytes written against the expected value
+     */
+
+    BIO *h = NULL;
+    char *buf = "011103";
+    char *expected_str;
+    char *obtained_str;
+
+    int expected;
+    int obtained;
+
+    TestResult* testResult = palloc0(sizeof(TestResult));
+    testResult->result = true;
+
+    h = BIO_new(BIO_s_mem());
+
+    prelogin = malloc(strlen(buf)/2 + 8);
+    expected = strlen(buf)/2;
+    expected_str = (char *)malloc(expected + 1);
+    strncpy(expected_str, buf, expected);
+    expected_str[expected] = '\0';
+
+    obtained = test_ssl_handshake_write(h, buf, expected, mock_socket_write);
+
+    obtained_str = (char *)malloc(obtained + 1);
+    strncpy(obtained_str, buf, obtained);
+    obtained_str[obtained] = '\0';
+
+    TEST_ASSERT_TESTCASE(expected == expected, testResult);
```

Comment:
nit: please add comment saying which ASSERT is checking what? I mean add comment saying this ASSERT checks the number of bytes, this ASSERT checks the actual byte message. Also, as we are checking the number of bytes, I think we should check the actual message only iff the number of bytes matches or else there is no need to write. So, better to put and if condition maybe.

Reply:
Updated in the next revision.